internal/kms/keyprotect.go

```diff
@@ -78,14 +94,20 @@ type KeyProtectKMS struct {
 	region            string
 	sessionToken      string
 	crk               string
+	providerName      string
 }
 
 func initKeyProtectKMS(args ProviderInitArgs) (EncryptionKMS, error) {
 	kms := &KeyProtectKMS{
 		namespace: args.Namespace,
 	}
+	// storing providername.
```

Comment:
This does not explain why the provider name needs to be stored. A comment about that would be helpful (in the struct too).

If this is only for the logging in `getSecrets()`, it would be sufficient to be clearer in the warning log and mention `"ibmkeyprotect"` is used.

Reply:
@nixpanic , I have kept only logging model now and mentioned in the log message as we are proceeding with new provider name.. should be good for now. ptal.